app/grandchallenge/archives/tasks.py

```diff
@@ -35,6 +38,34 @@ def add_images_to_archive(*, upload_session_pk, archive_pk, interface_pk=None):
         item.values.set([civ])
 
 
+@shared_task
+def add_images_to_archive_item(
+    *, upload_session_pk, archive_item_pk, interface_pk=None
+):
+    archive_item = ArchiveItem.objects.get(pk=archive_item_pk)
+    interface = ComponentInterface.objects.get(pk=interface_pk)
+
+    civ_pks_to_remove = set()
+    civ_pks_to_add = set()
+
+    civ = archive_item.values.filter(interface=interface).first()
```

Comment:
Only get the first? Archive items can only contain one of each interface type, so it should be safe to remove them all.

Reply:
Hm, maybe I'm missing something, but here is my reasoning for using `first()`: there can either be exactly 1 matching object, or None. So retieving `all()` is no differet than gettiing the first, right? The only diffference is that if I call `all()` I get a queryset back, which I would then need to loop over in order to add the civ.pk to `civ_pks_to_remove`. With `first()` I get the object directly, so it seemed easier this way. In any case, `first()` seemed better than `get()` because the latter would throw an error if there is no matching interface, `first()` in turn returns None in that case.

Reply:
There are no guarantees that it is one or `None`, and it is best to use this point to be really sure that there are no others to save errors from propagating. You should be able to do `archive_item.values.remove(*archive_item.values.filter(interface=interface))`, and `archive_item.values.add(new_civ)`.